src/bundle/ControllerArgumentResolver/ContentTreeChildrenQueryArgumentResolver.php

```diff
@@ -35,7 +34,7 @@ public function __construct(
 
     public function supports(Request $request, ArgumentMetadata $argument): bool
     {
-        return Criterion::class === $argument->getType()
+        return LogicalAnd::class === $argument->getType()
```

Comment:
I see what you're trying to achieve, but I don't think we can go in this direction. 

Since this argument resolver has been added, we added an expectation a developer can have towards application: using a `Criterion $filter` argument should be converted to a valid criterion.

So it has became a contract of sorts. It's actual usage is unlikely (I don't even know if it's documented), but it can happen.

I would suggest a different approach. Since what we're interested in is giving a valid Criterion, and since we don't have a "NullCriterion" that would indicate no conditions (and LogicalAnd expect at least one, otherwise it will fail), how about we check whether `$argument->isNullable()` in the `ContentTreeChildrenQueryArgumentResolver::resolve()` method - and yield `null` if we have an empty criteria array if so?

Then you'd remain compliant with what we already have, and you will even not need to do any changes to the controller code - apart from allowing `$filter` variable to be null.

Reply:
Yes, this approach is the best. I considered it at the beginning, but I didn't provide it because I thought we can't yield null values. PHPStan also reporting issue "Generator expects value type Ibexa\Contracts\Core\Repository\Values\Content\Query\Criterion, null given."

Reply:
Fixed: https://github.com/ibexa/admin-ui/pull/1328/commits/e1437833ba4c3c1d80dcc2b6794a9ab10a226e11